libcontainerd/container_linux.go

```diff
@@ -91,7 +91,8 @@ func (ctr *container) start(checkpoint string, checkpointDir string) error {
 	if err != nil {
 		return nil
 	}
-	iopipe, err := ctr.openFifos(spec.Process.Terminal)
+	createChan := make(chan struct{})
+	iopipe, err := ctr.openFifos(spec.Process.Terminal, createChan)
```

Comment:
How about re-wrapping it here instead(with a comment about why we need this hack): 
```
stdin := iopipe.Stdin
iopipe.Stdin = ioutils.NewWriteCloserWrapper(stdin, func() error {
  go func() {
    <-createChan
    stdin.Close()
   }()
  return nil
})
```